Paypal transactions aren't getting to Salesforce

One-time donations are the only ones working right now (see #514) and they show up in Braintree, but not Salesforce.

Oh! It seems to be choking on `first_name`

```
2019-11-01T02:13:47.314277+00:00 app[worker.1]: 2019-11-01 02:13:47,313 [ERROR] KeyError: 'first_name'
2019-11-01T02:13:47.314282+00:00 app[worker.1]: Traceback (most recent call last):
2019-11-01T02:13:47.314288+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.7/site-packages/rq/worker.py", line 822, in perform_job
2019-11-01T02:13:47.314294+00:00 app[worker.1]: rv = job.perform()
2019-11-01T02:13:47.314296+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.7/site-packages/rq/job.py", line 605, in perform
2019-11-01T02:13:47.314298+00:00 app[worker.1]: self._result = self._execute()
2019-11-01T02:13:47.314299+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.7/site-packages/rq/job.py", line 611, in _execute
2019-11-01T02:13:47.314301+00:00 app[worker.1]: return self.func(*self.args, **self.kwargs)
2019-11-01T02:13:47.314303+00:00 app[worker.1]: File "/app/donate/payments/tasks.py", line 43, in send_transaction_to_basket
2019-11-01T02:13:47.314304+00:00 app[worker.1]: 'first_name': data['first_name'],
2019-11-01T02:13:47.314306+00:00 app[worker.1]: KeyError: 'first_name'
2019-11-01T02:13:47.314307+00:00 app[worker.1]: Traceback (most recent call last):
2019-11-01T02:13:47.314309+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.7/site-packages/rq/worker.py", line 822, in perform_job
2019-11-01T02:13:47.314311+00:00 app[worker.1]: rv = job.perform()
2019-11-01T02:13:47.314312+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.7/site-packages/rq/job.py", line 605, in perform
2019-11-01T02:13:47.314314+00:00 app[worker.1]: self._result = self._execute()
2019-11-01T02:13:47.314315+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.7/site-packages/rq/job.py", line 611, in _execute
2019-11-01T02:13:47.314317+00:00 app[worker.1]: return self.func(*self.args, **self.kwargs)
2019-11-01T02:13:47.314318+00:00 app[worker.1]: File "/app/donate/payments/tasks.py", line 43, in send_transaction_to_basket
2019-11-01T02:13:47.314320+00:00 app[worker.1]: 'first_name': data['first_name'],
2019-11-01T02:13:47.314321+00:00 app[worker.1]: KeyError: 'first_name'
```

--- a/donate/payments/tests/test_views.py
+++ b/donate/payments/tests/test_views.py
@@ -14,6 +14,7 @@ from ..forms import (
     BraintreePaymentForm, BraintreeCardPaymentForm, BraintreePaypalPaymentForm,
     BraintreePaypalUpsellForm, UpsellForm
 )
+from ..tasks import send_transaction_to_basket
 from ..views import (
     BraintreePaymentMixin, CardPaymentView, CardUpsellView, NewsletterSignupView,
     PaypalPaymentView, PaypalUpsellView, TransactionRequiredMixin
@@ -23,6 +24,7 @@ from ..exceptions import InvalidAddress
 
 class MockPaypalDetails:
     payer_email = 'customer@example.com'
+    payer_first_name = 'Bob'
     payer_last_name = 'Jones'
 
 
@@ -47,6 +49,7 @@ class MockBraintreePaymentMethod:
     last_4 = '1234'
     email = 'test@example.com'
     payer_info = {
+        'first_name': 'Bob',
         'last_name': 'Jones'
     }
 
@@ -83,6 +86,7 @@ class BraintreeMixinTestCase(TestCase):
         view.request.session = {
             'landing_url': 'http://localhost',
             'project': 'thunderbird',
+            'campaign_id': 'piday',
         }
         view.request.LANGUAGE_CODE = 'en-US'
         view.success(MockBraintreeResult(), form, send_data_to_basket=False)
@@ -92,6 +96,7 @@ class BraintreeMixinTestCase(TestCase):
             'locale': 'en-US',
             'landing_url': 'http://localhost',
             'project': 'thunderbird',
+            'campaign_id': 'piday',
         })
 
 
@@ -348,6 +353,11 @@ class SingleCardPaymentViewTestCase(CardPaymentViewTestCase):
         })
         self.assertEqual(details, expected_details)
 
+        # Ensure that the details are valid for what the basket task expects
+        with mock.patch('donate.payments.tasks.send_to_sqs') as mock_send:
+            send_transaction_to_basket(self.view.prepare_session_data(details))
+        mock_send.assert_called_once()
+
 
 class MonthlyCardPaymentViewTestCase(CardPaymentViewTestCase):
 
@@ -415,6 +425,11 @@ class MonthlyCardPaymentViewTestCase(CardPaymentViewTestCase):
             'settlement_amount': None,
         })
         self.assertEqual(details, expected_details)
+
+        # Ensure that the details are valid for what the basket task expects
+        with mock.patch('donate.payments.tasks.send_to_sqs') as mock_send:
+            send_transaction_to_basket(self.view.prepare_session_data(details))
+        mock_send.assert_called_once()
 
 
 class PaypalPaymentViewTestCase(TestCase):
@@ -497,8 +512,9 @@ class PaypalPaymentViewTestCase(TestCase):
         self.view.currency = 'usd'
         result = MockBraintreeSubscriptionResult()
         payment_method = MockBraintreePaymentMethod()
+        details = self.view.get_transaction_details_for_session(result, form, payment_method=payment_method)
         self.assertEqual(
-            self.view.get_transaction_details_for_session(result, form, payment_method=payment_method),
+            details,
             {
                 'amount': Decimal(10),
                 'transaction_id': 'subscription-id-1',
@@ -507,9 +523,18 @@ class PaypalPaymentViewTestCase(TestCase):
                 'currency': 'usd',
                 'settlement_amount': None,
                 'email': 'test@example.com',
+                'first_name': 'Bob',
                 'last_name': 'Jones',
+                'campaign_id': '',
+                'project': 'mozillafoundation',
+                'landing_url': 'http://localhost',
             }
         )
+
+        # Ensure that the details are valid for what the basket task expects
+        with mock.patch('donate.payments.tasks.send_to_sqs') as mock_send:
+            send_transaction_to_basket(self.view.prepare_session_data(details))
+        mock_send.assert_called_once()
 
     def test_get_success_url_single(self):
         self.view.payment_frequency = 'single'

--- a/donate/payments/views.py
+++ b/donate/payments/views.py
@@ -53,15 +53,20 @@ class BraintreePaymentMixin:
         self.request.session['campaign_id'] = form.cleaned_data['campaign_id']
         self.request.session['project'] = form.cleaned_data['project']
 
-    def success(self, result, form, send_data_to_basket=True, **kwargs):
-        # Store details of the transaction in a session variable
-        details = self.get_transaction_details_for_session(result, form, **kwargs)
-        details.update({
+    def prepare_session_data(self, details):
+        data = {
             'locale': self.request.LANGUAGE_CODE,
             'landing_url': self.request.session.get('landing_url', ''),
             'project': self.request.session.get('project', ''),
-        })
-        details = freeze_transaction_details_for_session(details)
+            'campaign_id': self.request.session.get('campaign_id', ''),
+        }
+        data.update(details)
+        return freeze_transaction_details_for_session(data)
+
+    def success(self, result, form, send_data_to_basket=True, **kwargs):
+        # Store details of the transaction in a session variable
+        details = self.get_transaction_details_for_session(result, form, **kwargs)
+        details = self.prepare_session_data(details)
         self.request.session['completed_transaction_details'] = details
         if send_data_to_basket:
             queue.enqueue(send_transaction_to_basket, details)
@@ -364,19 +369,26 @@ class PaypalPaymentView(BraintreePaymentMixin, FormView):
             transaction_id = result.transaction.id
             settlement_amount = result.transaction.disbursement_details.settlement_amount
             email = result.transaction.paypal_details.payer_email
+            first_name = result.transaction.paypal_details.payer_first_name
             last_name = result.transaction.paypal_details.payer_last_name
         else:
             transaction_id = result.subscription.id
             settlement_amount = None
             email = kwargs['payment_method'].email
+            first_name = kwargs['payment_method'].payer_info['first_name']
             last_name = kwargs['payment_method'].payer_info['last_name']
+
         return {
             'amount': form.cleaned_data['amount'],
+            'campaign_id': form.cleaned_data['campaign_id'],
+            'project': form.cleaned_data['project'],
+            'landing_url': form.cleaned_data['landing_url'],
             'settlement_amount': settlement_amount,
             'transaction_id': transaction_id,
             'payment_method': constants.METHOD_PAYPAL,
             'currency': self.currency,
             'payment_frequency': self.payment_frequency,
+            'first_name': first_name,
             'last_name': last_name,
             'email': email,
         }

